### PR TITLE
[improve][fn] topic name is not required for source config

### LINKED ISSUE #15667: `topicName` should not be a `required` config for source connector
**Describe the bug**
For some of the source connectors, like debezium connector, it actually not sending any messages to the `topicName`. And the `topicName` cannot be null: https://github.com/apache/pulsar/blob/branch-2.10/pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/SourceConfigUtils.java#L296-L298

But from the java instance, if no topic is provided, it will use [PulsarSinkDisable](https://github.com/apache/pulsar/blob/branch-2.10/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSinkDisable.java) instead. 

We should remove the strict validation on `topicName` for the source connector.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #15667.

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/tisonkun/pulsar/pull/20

### CODE REVIEW

File: site2/docs/io-debezium-source.md [hunk 1]
```diff
@@ -170,15 +170,15 @@ This example shows how to change the data of a MySQL table using the Pulsar Debe
        ```bash
        bin/pulsar-admin source localrun \
        --archive connectors/pulsar-io-debezium-mysql-@pulsar:version@.nar \
-       --name debezium-mysql-source --destination-topic-name debezium-mysql-topic \
```
Comment (momo-jun) [inline site2/docs/io-debezium-source.md:173]: This subcommand also exists in `io-cdc-debezium.md`, which seems to be a duplicate of `io-debezium-source.md` but still referenced in `io-cdc.md`. Maybe not in this PR's scope but can be revisited.

Comment (tisonkun) [reply site2/docs/io-debezium-source.md:173]: @momo-jun Update `io-cdc-debezium.md`. For the duplications, let's handle them later.

**Review by freeznet — APPROVED** [review]

**Comment by tisonkun** [comment]
Merging...

**Pull request merged**